app/internal/page/data.go

```diff
@@ -161,42 +161,80 @@ func readDate(t time.Time) Date {
 	}
 }
 
+type WitnessCode struct {
+	Code    string
+	Created time.Time
+}
+
+func (w *WitnessCode) HasExpired() bool {
+	return w.Created.Before(time.Now().Add(-30 * time.Minute))
+}
+
 type LpaStore interface {
-	Get(context.Context, string) (*Lpa, error)
-	Put(context.Context, string, *Lpa) error
+	Create(context.Context) (*Lpa, error)
+	GetAll(context.Context) (*Lpa, error)
+	Get(context.Context) (*Lpa, error)
+	Put(context.Context, *Lpa) error
 }
 
 type lpaStore struct {
 	dataStore DataStore
 	randomInt func(int) int
 }
 
-type WitnessCode struct {
-	Code    string
-	Created time.Time
+type sessionData struct {
+	SessionID string
+	LpaID     string
 }
 
-func (w *WitnessCode) HasExpired() bool {
-	return w.Created.Before(time.Now().Add(-30 * time.Minute))
+func (d *sessionData) String() string {
+	if d == nil {
+		return "sessionData<nil>"
+	}
+
+	return fmt.Sprintf("sessionData(sessionID=%s lpaID=%s)", d.SessionID, d.LpaID)
+}
+
+func sessionDataFromContext(ctx context.Context) *sessionData {
+	data, _ := ctx.Value((*sessionData)(nil)).(*sessionData)
+
+	return data
+}
+
+func contextWithSessionData(ctx context.Context, data *sessionData) context.Context {
+	return context.WithValue(ctx, (*sessionData)(nil), data)
+}
+
+func (s *lpaStore) Create(ctx context.Context) (*Lpa, error) {
+	lpa := &Lpa{ID: "10" + strconv.Itoa(s.randomInt(100000))}
+	err := s.Put(ctx, lpa)
+
+	return lpa, err
 }
 
-func (s *lpaStore) Get(ctx context.Context, sessionID string) (*Lpa, error) {
+func (s *lpaStore) GetAll(ctx context.Context) (*Lpa, error) {
```

Comment:
I'm assuming this will return `[]*Lpa` on the next ticket?

Reply:
yep, that is the plan